### ISSUE TITLE
Clarification on limit parameter ...

### ISSUE THREAD

**Issue by pvretano** [opengeospatial/ogcapi-features#368]
Consider that a server has specified that the min and max values for the limit parameter are 10000.  If a client then makes a request for 50000 records should the server throw an exception (since 50000 is greater than 10000) or should it simply return 10000 records?

**Comment by nmtoken** [comment]
My gut feeling is it should return 10000 records.   

You could write the spec to say either must return, must not return, or may return... the latter a bit like in existing (WMS, WFS..) OGC specifications with duplicate params.

What behaviour is specified for requests larger than a specified bounding box or time period, that would be analogous.

**Comment by cholmes** [comment]
I agree it should return 10000 records, and then be sure to have numberReturned be 10000, so the client knows how many it got back. With numberMatched still reporting accurately, so they can page through if there's more.

**Comment by cportele** [comment]
Meeting 2020-04-27: Nothing to clarify, this is already what is said in
http://docs.opengeospatial.org/DRAFTS/17-069r4.html#_response_6 (see the third bullet):

> So (using the default/maximum values of 10/10000 ...):
> 
> * If you ask for 10, you will get 0 to 10 (as requested) and if there are more, a next link;
> * If you don’t specify a limit, you will get 0 to 10 (default) and if there are more, a next link;
> * If you ask for 50000, you might get up to 10000 (server-limited) and if there are more, a next link;
> * If you follow the next link from the previous response, you might get up to 10000 additional features and if there are more, a next link.

**Comment by jampukka** [comment]
What about the other way, say limit=-5, should you get 0 to 1 or 0 to 10 (default)? Using minimum/default values of 1/10. Or is this a bad request?

**Comment by tomkralidis** [comment]
Should we consider a `limit` (integer) property that conveys the default limit at the collection info (or landing page) level?

**Comment by jerstlouis** [comment]
@jampukka From how I read it, I believe it still does not require the server to consider the request _successful_ if the client request a larger limit than the server supports:

> If you ask for 50000, you **might** get up to 10000 (server-limited) 

I read this as meaning you might also net and get a 400 error.

> The server likely has a default value for the limit, and a maximum limit.

It's also not clear what the _maximum limit_ means. I believe it should be clarified that the server cannot return an error if the client request a limit larger than that _maximum limit_, but that it is the maximum number of results the server can return regardless of any limit specified by the client.

**Comment by jampukka** [comment]
Maybe we do need a clarification then. Following the same pattern of the first two bullets, would `If you ask for 50000, you will get 0 to 10000 (server declared maximum) and if there are more, a next link;` make it clear enough?

Maybe declaring minimum/default/maximum values only in the OpenAPI document is not the best solution especially as we require the server not to validate the parameter against its' own OpenAPI definition/schema. Somewhat related https://github.com/opengeospatial/ogcapi-features/issues/538

**Comment by joanma747** [comment]
This might be better:
```
C) The server SHALL return the maximum possible number of items if the limit parameter values greater than the one the server supports, rather than responding with an error.
```
